bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TupleValueImpl.java

```diff
@@ -827,4 +830,34 @@ private void resetSize(int index) {
             size = index + 1;
         }
     }
+
+    private void checkIfRestElementPresent (long length) {
+        if (this.tupleType.getTupleTypes().size() >= length) {
+            throw ErrorHelper.getRuntimeException(
+                    getModulePrefixedReason(ARRAY_LANG_LIB, OPERATION_NOT_SUPPORTED_IDENTIFIER),
+                    ErrorCodes.OPERATION_NOT_SUPPORTED_ERROR, "shift()", this.getType());
+        }
+    }
+
+    private void checkAdjacentElementInherentType () {
+        // Check if the i th member has a suitable value to replace the i-1 th member (Checking done by value, not type)
+        List<Type> tupleTypes = this.tupleType.getTupleTypes();
+        int numberOfTypes = tupleTypes.size();
+        for (int i = 1; i < numberOfTypes; i++) {
+            if (!TypeChecker.checkIsType(this.getRefValue(i), tupleTypes.get(i - 1))) {
+                throw ErrorHelper.getRuntimeException(
+                        getModulePrefixedReason(ARRAY_LANG_LIB, INHERENT_TYPE_VIOLATION_ERROR_IDENTIFIER),
+                        ErrorCodes.INCOMPATIBLE_TYPE, tupleTypes.get(i - 1), tupleTypes.get(i));
+            }
+            continue;
+        }
+        if (this.hasRestElement) {
+            if (!TypeChecker.checkIsType(this.getRefValue(this.getLength() - 1), tupleTypes.get(numberOfTypes - 1))) {
```

Comment:
We can merge the if condition here.

```suggestion
        if (this.hasRestElement && !TypeChecker.checkIsType(this.getRefValue(this.getLength() - 1), tupleTypes.get(numberOfTypes - 1))) {
```